components/details/DownloadDataLink.tsx

```diff
@@ -0,0 +1,63 @@
+import { Link, Text, Flex, VStack } from "@chakra-ui/react";
+import { memo } from "react";
+import { Entity } from "../../lib/model";
+import NextLink from "next/link";
+
+import { GrDocumentZip } from "react-icons/gr";
+
+export interface DownloadDataCSVProps {
+  entity: Entity;
+}
+
+const DownloadDataLink = ({ entity }: DownloadDataCSVProps) => {
+  const dataUrl = `${process.env.COKI_API_URL}/download/${entity.entity_type}/${entity.id}`;
+
+  // Request the ZIP
+  const fetchDataOnClick = async () => {
+    await fetch(dataUrl, {
+      method: "GET",
+      headers: {
+        Accept: "application/zip",
+      },
+    })
+      .then((res) => {
+        // Error handling incase the API does not give a successful response.
+        if (res.status === 200) {
+          return res.blob();
+        } else {
+          return undefined;
+        }
+      })
+      .then((data) => {
+        if (data != undefined) {
+          var a = document.createElement("a");
+          a.href = window.URL.createObjectURL(data);
+          a.download = `COKI_data_${entity.entity_type}_${entity.id}.zip`;
```

Comment:
Could we call the file Country name + COKI Dataset.zip, e.g "Mali COKI Dataset.zip"?